core/src/main/java/io/aiven/klaw/service/TopicSyncControllerService.java

```diff
@@ -630,6 +631,12 @@ private void createAndApproveTopicRequest(
       if (createResult.get("topicId") != null) {
         topicRequest.setTopicid(Integer.parseInt(createResult.get("topicId")));
         manageDatabase.getHandleDbRequests().updateTopicRequest(topicRequest, getUserName());
+        // After creating and approving this topic creation add the new topicId and environment as
+        // they are the only differences between the versions.
+        // This prevents us from having to do a dip into the DB to get the newly created topic.
+        topicFound.setEnvironment(syncBackTopics.getTargetEnv());
+        topicFound.setTopicid(topicRequest.getTopicid());
```

Comment:
within updateTopicRequest method, a new topic id is generated. I think that 'Topic' should be retrieved and pushed to cache